src/ja/server/database/sql/database.py

```diff
@@ -202,11 +203,13 @@ def find_job_by_label(self, label: str) -> List[Job]:
         jobs: List[Job] = session.query(Job).filter(Job.label == label).all()
         return jobs
 
-    def update_job(self, job: Job) -> None:
+    def update_job(self, job: Job) -> str:
         session = self.scoped()
         old_job_entry: Optional[DatabaseJobEntry] = self.find_job_by_id(job.uid)
         old_job = old_job_entry.job if old_job_entry else None
         if old_job is None:
+            if job.uid is None:
+                job.uid = str(job.owner_id) + str(int(time.time() * 1000))
```

Comment:
Please add `from pwd import getpwuid` to your imports and replace `str(job.owner_id)` with `getpwuid(job.owner_id).pw_name`.
That way job UIDs will be something like user12345 instead of just a string of numbers.